api_tests/nodes/views/test_node_detail.py

```diff
@@ -821,6 +824,38 @@ def test_can_make_project_public_if_admin_contributor(
             project_private.reload()
             assert project_private.is_public
 
+    def test_make_project_private(
+        self, app, url_public, project_public, user
+    ):
+        # If the public node exceeds the the private storage limit
+        key = cache_settings.STORAGE_USAGE_KEY.format(target_id=project_public._id)
+        storage_usage_cache.set(key, 7500000000, settings.STORAGE_USAGE_CACHE_TIMEOUT)
+        res = app.patch_json_api(url_public, {
+            'data': {
+                'type': 'nodes',
+                'id': project_public._id,
+                'attributes': {
+                    'public': False
+                }
+            }
+        }, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == 'This project exceeds private project storage limits and thus cannot be converted into a private project.'
+
+        # If the public node does not exceed the private storage limit
+        storage_usage_cache.set(key, 4900000000, settings.STORAGE_USAGE_CACHE_TIMEOUT)
```

Comment:
Probably should get the default storage cap, multiply by the GB setting, and subtract something to get this rather than using a number. That way, if the default cap changes, the test won't break.